### ISSUE TITLE
Hyperparameters about training SYMNet on 2D data.

### ISSUE THREAD

**Issue by MingR-Ma** [cwmok/Fast-Symmetric-Diffeomorphic-Image-Registration-with-Convolutional-Neural-Networks#9]
Hi, thanks for your work. I'm trying to train your SYMNet on the 2D slices. I used your hyperparameters setting(smooth=5, magnitude=0.001, local_ori=1000.0) and found the trained model result not good. Could you suggest to me how to set the hyperparameters to train SYMNet correctly?
![r1](https://user-images.githubusercontent.com/26457472/139616592-5405f84e-3ee4-4ac9-adf3-3fc4b4d884da.jpg)
![r2](https://user-images.githubusercontent.com/26457472/139616600-71d8ea1e-79bb-44f8-89af-2110c0dacdc7.jpg)

**Comment by cwmok** [comment]
Hi @MingR-Ma, Regarding the hyperparameter tuning technique, you may refer to my reply in #6. Yet, I think the problem is in your dataset. It seems that the target structure has zero intensity at most locations and it will be regarded as background intensity in the default similarity function. I think the best solution is to replace the default NCC similarity function with the Dice similarity of the label between fixed image and moving image.  

Have you tried other registration methods? Are they performing well in an unsupervised fashion? If yes, use the same similarity metric for SYMNet and it will work.

**Comment by MingR-Ma** [comment]
Thanks for your reply. As I know, the performance of NCC on 2D data is worse than MSE. I have trained VM and vit-v-net (they use MSE as loss function), they got similar results. If I use MSE loss function, what hyperparameters should I use? The VM's prediction is below.
![r1](https://user-images.githubusercontent.com/26457472/139631474-65f11909-8e7d-4443-8c81-8161e53a3a22.jpg)
![r2](https://user-images.githubusercontent.com/26457472/139631476-4f59ef3a-43ee-46dc-8348-edd2ec02fa0f.jpg)

**Comment by cwmok** [comment]
> As I know, the performance of NCC on 2D data is worse than MSE.

Yes, because NCC considers pixels with 0 intensity as background.

> If I use MSE loss function, what hyperparameters should I use?

I haven't tried it before, but I think setting the lambda to match the lambda you used in VM will be fine. But I don't think the result will be good in your case. Because both similarity metrics cannot really quantify the similarity of the targeting structure **_at every pixel location._** As you know deformable registration aims to build up a 1-to-1 correspondence between two images. If there are no quantified similarities of the targeting structure **_at every pixel location_**, both VM and our method cannot learn the expansion and contraction deformation of the targeting structure very well.

On top of my head, I have two solutions for your case. 
1. Use Dice loss instead.
2. Use a deformation model with stronger regularization likes B-spline or grid with control points to avoid direct dense displacement field estimation.

**Comment by MingR-Ma** [comment]
Very inspired reply! First, I will test MSE on SYMnet, then try the dice loss. I will reply to you @cwmok  later.
